### DISCUSSION TITLE
GitHub Copilot extension for Visual Studio not triggering inside of .razor files

### DISCUSSION THREAD

**Discussion by EngstromJimmy** [community/community#12594] [Copilot Conversations] [answered]
When using GitHub Copilot extension for Visual Studio (2022) it doesn't work from inside @code{} inside of .razor files.
It simply doesn't activate.
Workaround:
Use .cs files but once you go .razir you never want to go back =)

**Comment by hsenag** [comment]
Thanks for the report! We'll take a look soon.

**Comment by kzu** [comment]
I'm able to successfully use it. Could you provide a repro, perhaps a recording?

![image](https://user-images.githubusercontent.com/169707/158638172-9cc958a7-a9ae-4af2-9809-efbf5ff8119f.png)

**Comment by EngstromJimmy** [comment]
Strange
This is what I get:
![Razor files](https://user-images.githubusercontent.com/2390244/158992704-0215e83d-7896-4a27-b701-58d92c319afa.gif)
This is what I expect to happen (this is from a class file).
![Expected](https://user-images.githubusercontent.com/2390244/158992793-93b7740c-44d4-456e-abce-f13d27d06285.gif)

Could it be the help you are seeing is from Intellicode and not from CoPilot?

**Reply by kzu** [reply]
This most definitely is *not* Intellicode:

![razorworks](https://user-images.githubusercontent.com/169707/159048211-a562fc75-7450-4ede-be8f-87b426de2817.gif)

That said, triggering does not seem to happen from within the `()`, unless I force trigger by hitting `Alt + ,` (or alternatively, `Ctrl + Alt + \`)...

![razortrigger](https://user-images.githubusercontent.com/169707/159048735-41171cce-d350-4c34-a6c3-dd2ec5f56397.gif)

**Reply by kzu** [reply]
Could you please provide your Help | About Visual Studio (use Copy Info)?

**Comment by kzu** [comment]
Keep in mind you can always force-trigger a suggestion by pressing `Ctrl+Alt+\`, which helps when completion/intellisense is otherwise spotty (as in the case of the parenthesis above, even the built-in intellisense isn't triggering automatically)

**Comment by EngstromJimmy** [comment]
I upgraded my Visual Studio, I was running an old preview.
It seems to work better now, the behavior isn't exactly the same.
When I type ...FizzBuzz() and then enter/return I get Co-pilot to trigger.

So from my perspective, it works as you describe it, I need to trigger the suggestion.
Didn't get alt +, to work though (VS plays a sound and nothing happens) and Ctrl + alt + \ is almost impossible to hit using a Swedish keyboard.

Microsoft Visual Studio Enterprise 2022
Version 17.2.0 Preview 2.0
VisualStudio.17.Preview/17.2.0-pre.2.0+32314.265
Microsoft .NET Framework
Version 4.8.04161

Installed Version: Enterprise

Visual C++ 2022   00476-80000-00000-AA212
Microsoft Visual C++ 2022

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ASP.NET and Web Tools 2019   17.2.240.24236
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   17.2.240.24236
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.2.240.24236
Azure Functions and Web Jobs Tools

C# Tools   4.2.0-2.22159.10+f3a5bad242b7a7b8149ae644de0a61c2f1bffc8d
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub Copilot   1.7.6029.0 (v1.7.29.0@46d50adc1)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.7.5227 (v1.7.29)

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

NuGet Package Manager   6.2.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2215202+658e2db05d0daa9326356ee1edc85cd3d1b970c4
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62203.01080
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.10302.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.2.0-2.22159.10+f3a5bad242b7a7b8149ae644de0a61c2f1bffc8d
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.22108.1+63bcc94efda4fe19b612b167a25fdbcee796040b
Microsoft Visual F# Tools

Visual Studio Inline Suggestions   2.2.1910.61968
Ghost text API for Visual Studio inline suggestions

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

**Comment by EngstromJimmy** [comment, marked as answer]
Ok, apparently Ctrl + alt + § is the default mapping on a Swedish machine for Edit.Copilot.TriggerInlineSuggesion.
Added alt + ,  as shortcut and it works great.

**Reply by Vendetta3337** [reply]
Top super best

**Comment by ddlaat** [comment]
I had to turn on 'Show completions on new lines' at the little purple icon on the bottom of the IDE to trigger Copilot automatically:

![image](https://github.com/community/community/assets/31969132/d8f34028-c29d-46d0-a7bc-2edde85d08ed)
